### PR TITLE
deleting thread now redirects to discussions

### PR DESCRIPTION
**Pull request by Israellund**
## Link to Issue
Closes: #8744 

## Description of Changes
- User is now redirected to `/discussions` after deleting a thread from within the thread

## "How We Fixed It"

-created a new function `getThreadUrlPath` to replace the use of `getProposalUrlPath` that redirects to `/discussions` instead of `/proposal/${id}` if `type === ProposalType.Thread` is false
NOTE: I changed `getProposalUrlPath` to `getThreadUrlPath` in `DiscussionsPage.tsx` as well as `ViewThreadPage.tsx` to stay consistent with coding intent, despite it not being needed in `DiscussionsPage.tsx` 
## Test Plan
- go to a thread you create or create a thread
- while on `/discussions` page delete the thread
- confirm that thread was deleted and that you are still on `/discussions`
- go to another thread you created or create a new thread
- click into that thread and delete the thread from there
- confirm that you are now redirected to `/discussions`

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx [hunk 1]
```diff
@@ -127,6 +127,29 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
...
+    () => decodeURIComponent(splitURLPath[3]),
+    [splitURLPath],
+  );
+
+  //redirects users to All Discussions if they try to access a topic in the url that doesn't exist
+  useEffect(() => {
+    if (
+      !isLoadingTopics &&
+      decodedString &&
+      splitURLPath[2] === 'discussions' &&
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:141]: Did we test this piece against custom domains?

Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:141]: @jnaviask This should be fine for custom domains as the url is being split on "/", so there should be no difference.

Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:141]: in https://gov.1inch.io/discussions 
image


in https://commonwealth.im/1inch/discussions
image

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx [hunk 2]
```diff
@@ -127,6 +127,29 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
...
+      splitURLPath.length === 4
+    ) {
+      const validTopics = topics?.some(
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:144]: why you check topics here? The bug is about deleted thread, not topic.

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx (hunk 1 above)
Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:141]: Whoops. Okay, I was wrong, I will fix that.

**Review by masvelio — CHANGES_REQUESTED** [review]
infinite loader after visiting deleted thread 
https://github.com/user-attachments/assets/711a3159-f18c-4c2b-98d7-4760ebe535bc

This fix seems to be unrelated to the bug. We should check if thread data is fetched in ViewThreadPage. If not, redirect the user

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx (hunk 2 above)
Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:144]: I don't know why this code was included in this PR, this is from a different PR I had made to redirect users if they input the incorrect url that was merged already. But I will now change it here to account for custom domains, as mentioned above.

**Comment by Israellund** [comment]
@masvelio I changed the logic to handle custom urls and pushed that change. I wasn't sure how to replicate what you were showing in your loom, as that is what was shown before when a thread was deleted from inside the thread. I tried to create a thread, copy the url, delete the thread, then paste in the now deleted thread url, but it just shows "Thread not found". It also shows this if I delete the topic name after the "-" 
Please let me know if I'm missing something in regards to your loom though.

**Comment by burtonator** [comment]
@Israellund I went ahead and approved but there is this issue which is further complicated by this fix:

https://github.com/hicommonwealth/commonwealth/issues/8853

Basically we have too many ways to redirect users to URLs and half of them are inconsistent.

**Comment by Israellund** [comment]
@masvelio I used  // eslint-disable-next-line react-hooks/exhaustive-deps for the useEffect dependency array because if navigate is included in it it causes an infinite rerender. You commented on another PR that we shouldn't be using this, so let me know if you'd like for me to change it at all. This was the best way I could find to fix this.

**Comment by Israellund** [comment]
This PR handles the bad url redirect better as it accounts for custom domains, therefore I am closing [this PR](https://github.com/hicommonwealth/commonwealth/pull/8729)

**Comment by Israellund** [comment]
Closing this PR. The bug found when deleting a thread was fixed in #8248 and the custom domains fix mentioned in the comments above is fixed in PR #8928

**Pull request closed without merging**